internal/xds/resolver/xds_resolver.go

```diff
@@ -590,7 +590,7 @@ func (r *xdsResolver) onRouteConfigResourceUpdate(name string, update xdsresourc
 		return
 	}
 
-	r.applyRouteConfigUpdate(update)
+	r.applyRouteConfigUpdate(*update)
```

Comment:
We should change `applyRouteConfigUpdate` to accept a pointer and change the `currentRouteConfig` field to hold a pointer as well. That we won't have to deref the pointer here.